app/services/teachers/change_schedule.rb

```diff
@@ -9,23 +9,40 @@ def initialize(author:, lead_provider:, teacher:, training_period:, schedule:, s
       @schedule = schedule
       @school_partnership = school_partnership
       @author = author
+      @original_schedule = training_period&.schedule
     end
 
     def change_schedule
       ActiveRecord::Base.transaction do
-        finished_on = determine_finished_on
-
         track_payments_frozen_year!
-        finish_training_period!
-        new_training_period = create_new_training_period!(finished_on:)
-        record_change_schedule_event!(original_training_period: training_period, new_training_period:)
+
+        if training_period.started_on.past?
+          create_new_training_period!
+        else
+          update_training_period_in_place!
+        end
       end
 
       teacher
     end
 
   private
 
+    def update_training_period_in_place!
+      training_period.update!(schedule:, school_partnership:, expression_of_interest: school_partnership.active_lead_provider)
```

Comment:
I think we may only want to set `expression_of_interest` if the training period currently has one